src/world/lazy.rs

```diff
@@ -252,21 +254,69 @@ impl LazyUpdate {
     where
         F: FnOnce(&World) + 'static + Send + Sync,
     {
-        self.queue.0.push(Box::new(f));
+        self.queue.as_ref().unwrap().0.push(Box::new(|w: &mut World| f(w)));
     }
 
-    pub(super) fn maintain(&mut self, world: &World) {
-        let lazy = &mut self.queue.0;
+    /// Lazily executes a closure with mutable world access.
+    ///
+    /// This can be used to add a resource to the `World` from a system.
+    ///
+    /// ## Examples
+    ///
+    /// ```
+    /// # use specs::prelude::*;
+    /// #
+    ///
+    /// struct Sys;
+    ///
+    /// impl<'a> System<'a> for Sys {
+    ///     type SystemData = (Entities<'a>, Read<'a, LazyUpdate>);
+    ///
+    ///     fn run(&mut self, (ent, lazy): Self::SystemData) {
+    ///         for entity in ent.join() {
+    ///             lazy.exec_mut(move |world| {
+    ///                 // complete extermination!
+    ///                 world.delete_all();
+    ///             });
+    ///         }
+    ///     }
+    /// }
+    /// ```
+    pub fn exec_mut<F>(&self, f: F)
```

Comment:
Do we really need separate mutable execute method? Can't we just promote normal execute to this? Just wondering.

Reply:
I think I used this to make this non-breaking.